pyiceberg/expressions/parser.py

```diff
@@ -78,6 +78,8 @@
 identifier = Word(alphas, alphanums + "_$").set_results_name("identifier")
 column = DelimitedList(identifier, delim=".", combine=False).set_results_name("column")
 
+like_regex = r'(?P<valid_wildcard>(?<!\\)%$)|(?P<invalid_wildcard>(?<!\\)%)'
```

Comment:
My first choice would not be a regex, but it works well 👍

Reply:
I didn't want to complicate the parser, so this seemed like the most straightforward path.